### ISSUE TITLE
Empty object and assembly files for simple assignment program (free and fixed format)

### ISSUE THREAD

**Issue by carlobertolli** [llvm-flang/flang-old#3]
Flang generates a LLVM-IR file that looks fine.

In free-format:

program main
  integer*4 bla
  bla = 3
end program main

**Comment by jleidel** [comment]
Carlo, I've verified that bug.  I'm triaging as well.

**Comment by jleidel** [comment]
fixed in merge cb275f9213b4e3173be2cd6282bd8196c55910b6

**Comment by carlobertolli** [comment]
Merged locally, but I obtain this:

#0 0x12401a40 llvm::sys::PrintStackTrace(llvm::raw_ostream&) /gsa/yktgsa/projects/l/llvm-patch-review/repo/src/lib/Support/Unix/Signals.inc:430:15
#1 0x12402a3c PrintStackTraceSignalHandler(void*) /gsa/yktgsa/projects/l/llvm-patch-review/repo/src/lib/Support/Unix/Signals.inc:488:3
#2 0x12402da8 SignalHandler(int) /gsa/yktgsa/projects/l/llvm-patch-review/repo/src/lib/Support/Unix/Signals.inc:206:5
0  flang     0x0000000012401a40
1  flang     0x0000000012402a3c
2  flang     0x0000000012402da8
3            0x00003fff9f110478 __kernel_sigtramp_rt64 + 0
4  flang     0x0000000010033db0
5  flang     0x0000000010033d3c
6  flang     0x0000000010033b58
7  flang     0x00000000100335b0
8  flang     0x0000000010032688
9  libc.so.6 0x00003fff9ebd4e80
10 libc.so.6 0x00003fff9ebd5074 __libc_start_main + 196

Input program is:

```
  program foo
  dimension a(10)
  real a
  dimension b(10)
  real b
  dimension c(10)
  real c
  integer i

  do 100 i=1,10
     a(i) = 1
     b(i) = i
```

 100  continue

```
  do 200 i=1,10
     c(i) = a(i) + b(i)
```

 200  continue

```
  do 300 i=1,10
     print *, i, ' and ', c(i)
```

 300  continue

```
  end
```
